pkg/api/v1/schema.go

```diff
@@ -115,6 +115,31 @@ type CloudInitNoCloudSource struct {
 	NetworkData string `json:"networkData,omitempty"`
 }
 
+// Represents a cloud-init config drive user data source.
```

Comment:
What about making the data types generic and the source type a "driver" for building the requested data source? This would result in less duplication of code and better reuse of similar code every time a new source type is added.

Reply:
I like that Idea very much, and in the cloud-init package I implemented such a generic data type (named `CloudInitData`) that minimises code duplication after the cloud-init data was read from the VMI schema.
But I am unsure how to implement the generic data type for the VMI schema. I think two things are relevant for such a generic data type:

1. In order to be backwards compatible, we need to keep `CloudInitNoCloudSource` around.
2. We need to consider that not all cloud-init data sources are volume-based. If KubeVirt decides to add a metadata service, cloud-init data could also come from that service.

So something like a `cloudInit` attribute in the `VirtualMachineInstanceSpec` might make sense...